File: HomeServer/dao/moduledao.py
from config.databaseconfig import *
from model.module import Module


class ModuleDao(object):

    @db_session
    def create_module(self, name, ip_address):
        try:
            return Module(name=name, ip_address=ip_address)
        except ValueError:
            rollback()

    @db_session
    def read_modules(self):
        return Module.select()[:]

    @db_session
    def read_module(self, module_id):
        return Module[module_id]

    @db_session
    def update_module(self, module_id, name, ip_address):
        m = Module[module_id]
        m.name = name
        m.ip_address = ip_address
        return m

    @db_session
    def delete_module(self, module_id):
        Module[module_id].delete()

File: MobileApplication/app/src/main/java/pl/polsl/mateusz/chudy/mobileapplication/dao/ModuleDao.kt
package pl.polsl.mateusz.chudy.mobileapplication.dao

import android.arch.persistence.room.*
import pl.polsl.mateusz.chudy.mobileapplication.model.Module

/**
 *
 */
@Dao
interface ModuleDao {

    @Query("select * from module")
    fun getModules(): List<Module>

//    @Query("select * from module where module_id = :moduleId")
//    fun getModule(moduleId: Long): Module
//
//    @Insert
//    fun createModule(module: Module)
//
//    @Update
//    fun updateModule(module: Module)

//    @Query("delete from module where module_id = :moduleId")
//    fun deleteModule(moduleId: Long)

    @Delete
    fun deleteModule(module: Module)
}